fuzzers/utils.py

```diff
@@ -108,3 +109,36 @@ def restore_directory(directory):
             os.getcwd()
         except FileNotFoundError:
             os.chdir(initial_cwd)
+
+
+def get_dictionary_path(target_binary):
+    """Return dictionary path for a target binary."""
+    if os.getenv('SKIP_DICT'):
```

Comment:
If not all the fuzzers support using dictionaries, would it make sense to make not using them the deafult, and only use them if explicitly requested (eg with env var)? I'm worried about fairness for fuzzers that don't take dicts.

Reply:
This is a tough one, right now all fuzzers have dicts except eclipser [dicts is way more common feature right now, and it can change in future]. And would this mean we don't allow fuzzers to add benchmark specific optimizations other than seed corpus? As an example, AFLSmart provides pits for file formats for some benchmarks, we allow this and i feel this is ok [i think this will let fuzzers try things other than dumb/coverage guided mutations]. We can let community provides ideas here too.

Reply:
I think we can allow these optimizations and do fair evaluation at the same time. I agree that it's very useful to measure the effect of various benchmark specific optimizations, such as a dictionary file or an input format spec. However, it's also important that we compare apples to apples. So if we have fuzzers that doesn't support dicts or pits, we should only use those as a separate configs. For example, it would be unfair to compare afl-smart that uses input specs with afl that can't use them -- as we'd be giving the former additional information (provided by a human) to work with. Instead, we should compare "afl-smart-pit", "afl-smart", and "afl" in that case. Does that make sense?

Reply:
This is a fair point. At least in the case of dictionaries though, they are so ubiquitous that I think performance with dictionaries is pretty much as interesting as performance without, it doesn't seem like such a "special case" in my mind.

I don't really know what to do about aflsmart here. I think as long as it uses prewritten pits that aren't tuned once added to Fuzzbench it's probably fine to include in the generic report. It doesn't seem that different from a fuzzer that infers the format of a corpus and does special mutations based on the format (I think radamsa does this for XML). If it starts getting tuned to fuzzbench then it's more of a problem (writing a new pit for a common format would be more OK IMO). Need to think more on this.

Reply:
Just to clarify: I wasn't arguing against including any of these. I was arguing for including *both* "with dict"/"without dict" (or "with pit"/"without pit") configs *if* not all fuzzers support it. This allows us to (1) see the effect of the dict or pit itself (eg. "afl-smart-pit" <> "afl-smart"), and (2) compare the raw effectiveness of the engine itself (eg. "afl-smart" <> "afl"). 
In cases when all fuzzers support them, we don't need to include the "without" versions.